tools/pre_ci.py

```
@@ -0,0 +1,486 @@
+        print("=" * 60, flush=True)
+
+        # 1. Sync Dependencies (Fail-fast as subsequent steps depend on it)
+        self.run_command(
+            ["uv", "sync", "--all-extras", "--frozen"],
+            "Syncing Project Environment",
+            fail_fast=True,
+        )
+
+        # 2. Update README
+        print("\n>>> [Step: Updating README structure]", flush=True)
+        res = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)